src/tools/distillery.ml

```diff
@@ -207,19 +211,29 @@ let preds () = [
 ]
 
 
-let get_templatedir () = Findlib.package_directory "eliom.templates"
+let get_templatedirs () =
+  let distillery_path_dirs =
+    try
+      let paths = Sys.getenv "ELIOM_DISTILLERY_PATH" in
+      Str.split (Str.regexp ":") paths
+    with
+    | Not_found -> []
+  in
+  eliom_template_dir::distillery_path_dirs
 
 let get_templates () =
-  let dir = Unix.opendir (get_templatedir ()) in
-  let rec aux rl =
+  let dirs = List.map (fun d -> (d, Unix.opendir d)) (get_templatedirs ()) in
+  let rec aux rl (path, dir) =
     try
       let f = Unix.readdir dir in
       if f = ".." || f = "."
-      then aux rl
-      else aux (f::rl)
+      then aux rl (path, dir)
+      else
+        aux ((f, path)::rl) (path, dir)
     with
       | End_of_file -> Unix.closedir dir; rl
-  in aux []
+  in List.concat (List.map (aux []) dirs)
```

Comment:
I suspect all this would be cleaner with a `fold_left` over `get_templatedirs ()` (with the `opendir` and concatenation happening inside), but no big deal.